DuckDuckGoTests/TabURLInterceptorTests.swift

```diff
@@ -0,0 +1,66 @@
+//
+//  TabURLInterceptorTests.swift
+//  DuckDuckGo
+//
+//  Copyright © 2024 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import XCTest
+import Subscription
+@testable import DuckDuckGo
+
+class TabURLInterceptorDefaultTests: XCTestCase {
+    
+    var urlInterceptor: TabURLInterceptorDefault!
+    
+    override func setUp() {
+        super.setUp()
+        urlInterceptor = TabURLInterceptorDefault()
+    }
+    
+    override func tearDown() {
+        urlInterceptor = nil
+        super.tearDown()
+    }
+    
+    func testAllowsNavigationForNonDuckDuckGoDomain() {
+        let url = URL(string: "https://www.example.com")!
+        XCTAssertTrue(urlInterceptor.allowsNavigatingTo(url: url))
+    }
+    
+    func testAllowsNavigationForUninterceptedDuckDuckGoPath() {
+        let url = URL(string: "https://duckduckgo.com/about")!
+        XCTAssertTrue(urlInterceptor.allowsNavigatingTo(url: url))
+    }
+    
+    func testNotificationForInterceptedPrivacyProPath() {
+        let expectation = self.expectation(forNotification: .urlInterceptPrivacyPro, object: nil, handler: nil)
+        
+        let url = URL(string: "https://duckduckgo.com/pro")!
+        let canNavigate = urlInterceptor.allowsNavigatingTo(url: url)
+        
+        // Simulate purchase allowance
+        SubscriptionPurchaseEnvironment.canPurchase = true
```

Comment:
Not for this PR but this kind of global state will not work in future

Reply:
yea.. sadly there's no clear way to mock or substitute for testing on this one.  We could wrap in protocol for this test, but this should probably be rewritten as part of the Unit testing efforts.